integration-tests/test/rpc.spec.ts

```diff
@@ -0,0 +1,172 @@
+
+      expect(result.from).to.equal(wallet.address)
+      expect(result.nonce).to.equal(nonce)
+      expect(result.gasLimit.toNumber()).to.equal(tx.gasLimit)
+      expect(result.gasPrice.toNumber()).to.equal(tx.gasPrice)
+      expect(result.data).to.equal(tx.data)
+    })
+
+    it('should not accept a transaction with the wrong chain ID', async () => {
+      const tx = {
+        ...defaultTransactionFactory(),
+        gasPrice: await gasPriceForL2(),
+        chainId: (await wallet.getChainId()) + 1,
+      }
+
+      await expect(
+        env.l2Provider.sendTransaction(await wallet.signTransaction(tx))
+      ).to.be.rejectedWith('invalid transaction: invalid sender')
+    })
+
+    it('should accept a transaction without a chain ID', async () => {
+      const tx = {
+        ...defaultTransactionFactory(),
+        nonce: await wallet.getTransactionCount(),
+        gasPrice: await gasPriceForL2(),
+        chainId: null, // Disables EIP155 transaction signing.
+      }
+      const signed = await wallet.signTransaction(tx)
+      const response = await env.l2Provider.sendTransaction(signed)
+
+      expect(response.chainId).to.equal(0)
+      const v = response.v
+      expect(v === 27 || v === 28).to.be.true
+    })
+
+    it('should accept a transaction with a value', async () => {
+      const tx = {
+        ...defaultTransactionFactory(),
+        gasPrice: await gasPriceForL2(),
+        chainId: await env.l2Wallet.getChainId(),
+        data: '0x',
+        value: ethers.utils.parseEther('0.1'),
+      }
+
+      const balanceBefore = await env.l2Provider.getBalance(
+        env.l2Wallet.address
+      )
+      const result = await env.l2Wallet.sendTransaction(tx)
+      const receipt = await result.wait()
+      expect(receipt.status).to.deep.equal(1)
+
+      const balAfter = await env.l2Provider.getBalance(env.l2Wallet.address)
+      expect(balAfter.lte(balanceBefore.sub(ethers.utils.parseEther('0.1')))).to
+        .be.true
+    })
+
+    it('should reject a transaction with higher value than user balance', async () => {
+      const balance = await env.l2Wallet.getBalance()
+      const tx = {
+        ...defaultTransactionFactory(),
+        gasPrice: await gasPriceForL2(),
+        chainId: await env.l2Wallet.getChainId(),
+        data: '0x',
+        value: balance.add(ethers.utils.parseEther('1')),
+      }
+
+      await expect(env.l2Wallet.sendTransaction(tx)).to.be.rejectedWith(
+        'invalid transaction: insufficient funds for gas * price + value'
+      )
+    })
+
+    it('should correctly report OOG for contract creations', async () => {
+      const factory = await ethers.getContractFactory('TestOOGInConstructor')
+
+      await expect(factory.connect(wallet).deploy()).to.be.rejectedWith(
+        'gas required exceeds allowance'
+      )
+    })
+  })
+
+  describe('debug_traceTransaction', () => {
+    before(async function () {
+      if (!envConfig.RUN_DEBUG_TRACE_TESTS) {
```

Comment:
I think we can remove this flag, since we always want to run debug trace tests now.